src/services/projectService.ts

```diff
@@ -230,6 +233,8 @@ export default class ProjectService implements IProjectService {
             });
             if (project.tags) {
                 project.tags = patch(project.tags, tags, "name", ["type", "format"]);
+                const newTags = tags.filter((fileTag) => !project.tags.find((tag) => fileTag.name === tag.name ));
```

Comment:
Will we assign color to new tag property?

Reply:
yes, in the setColorsForUpdatedTags function, a project's existing tag's color wont change and all new tags will get the next color to be used from tagColors.json